src/axom/sidre/core/View.hpp

```diff
@@ -862,10 +871,12 @@ class View
   {
     if ( !isAllocated() || !isDescribed())
     {
-      SLIC_CHECK_MSG( isAllocated(),
-                      "No view data present, memory has not been allocated.");
-      SLIC_CHECK_MSG( isApplied(),
-                      "View data description not present.");
+      SLIC_CHECK_MSG(isAllocated(),
+                     SIDRE_VIEW_LOG_PREPEND
+                     << "No view data present, memory has not been allocated.");
+      SLIC_CHECK_MSG(isDescribed(),
```

Comment:
Can someone please confirm that this change is correct? It seems correct to me.

Note that the outer conditional and the comment relate to ``isDescribed()``, but the inner check was on ``isApplied()``.

Tag: @rhornung67

Reply:
Seems like the outer conditional is superfluous, since the SLIC macros evaluate the same conditions?  These are warning checks, so the correct choice is whatever information we decide merits a warning.  I think that all three possibilities of unallocated, unapplied, and undescribed deserve a warning, as any of them indicate that the View may not be fully ready for use.

Reply:
Thanks @nselliott.

In this case, it looks like the outer ``if``s are superfluous since there is no additional behavior in the conditional.
They are also potentially expensive since the inner ``SLIC_CHECK_MSG`` can be compiled out, but the outer checks in the ``if`` are always applies.

I will remove them.